homeassistant/components/maxcube/__init__.py

```diff
@@ -112,3 +118,80 @@ def update(self):
                 self._updatets = time.monotonic()
             else:
                 _LOGGER.debug("Skipping update")
+
+    def disconnect(self):
+        """Disconnect from cube."""
+        with self.mutex:
+            self.cube.disconnect()
+
+
+async def _setup(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up using flow."""
+    config = entry.data
+
+    if DATA_KEY not in hass.data:
+        hass.data[DATA_KEY] = {}
+
+    # Get configs
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+    scan_interval = config[CONF_SCAN_INTERVAL]
+
+    if host in hass.data[DATA_KEY]:
+        # Already configured, do nothing
+        _LOGGER.debug("Already configured, do nothing")
+        return True
+
+    try:
+        cube = MaxCube(host, port, now=now)
+        hass.data[DATA_KEY][host] = MaxCubeHandle(cube, scan_interval)
+    except timeout as ex:
+        _LOGGER.error("Unable to connect to Max!Cube gateway: %s", str(ex))
+        persistent_notification.create(
+            hass,
+            f"Error: {ex}<br />You will need to reload integration after fixing.",
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID,
+        )
+        return False
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, Platform.CLIMATE)
+    )
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, Platform.BINARY_SENSOR)
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up config entry."""
+    res = await _setup(hass, entry)
+
+    entry.add_update_listener(async_reload_entry)
+    return res
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(
+        entry, [Platform.CLIMATE, Platform.BINARY_SENSOR]
+    )
+
+    host = entry.data[CONF_HOST]
+
+    if host not in hass.data[DATA_KEY]:
+        # Do nothing
+        return unload_ok
+
+    handler = hass.data[DATA_KEY].pop(host)
+    handler.disconnect()
+
+    return unload_ok
+
+
+async def async_reload_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
```

Comment:
You may be able to remove this using `hass.config_entries.async_reload(entry.entry_id)` instead

Reply:
1745a78bdf20cef356e2ccc5a7173f950ac75772